ts/private/ts_lib.bzl

```diff
@@ -87,6 +87,18 @@ https://docs.aspect.build/rulesets/aspect_rules_js/docs/js_library#deps for more
         mandatory = True,
         allow_single_file = [".json"],
     ),
+    "isolated_typecheck": attr.bool(
+        doc = """\
+        Whether type-checking should be a separate action.
+
+        This allows the transpilation action to run without waiting for typings from dependencies.
+
+        Requires the typescript 5.6 [noCheck](https://www.typescriptlang.org/tsconfig#noCheck).
```

Comment:
"Requires a minimum version of typescript 5.6 for the [noCheck](https://www.typescriptlang.org/tsconfig#noCheck) flag which is automatically set on the transpilation action when the typecheck action is isolated."